examples/nucleo_h723zg/bmi088/spi/main.cpp

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (c) 2023, Christopher Durand
+ *
+ * This file is part of the modm project.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+// ----------------------------------------------------------------------------
+
+#include <modm/driver/inertial/bmi088.hpp>
+
+#include <modm/board.hpp>
+#include <atomic>
+
+using namespace Board;
+
+using Spi = SpiMaster2_Dma<Dma1::Channel0, Dma1::Channel1>;
+using CsGyro = GpioC0;
+using CsAcc = GpioD6;
+using Mosi = GpioC3;
+using Miso = GpioC2;
+using Sck = GpioD3;
+
+using AccInt1 = GpioC8;
+using GyroInt3 = GpioC9;
+
+using Transport = modm::Bmi088SpiTransport<Spi, CsAcc, CsGyro>;
+using Imu = modm::Bmi088<Transport>;
+Imu imu;
+
+void initializeImu()
+{
+	AccInt1::setInput(AccInt1::InputType::PullDown);
+	GyroInt3::setInput(GyroInt3::InputType::PullDown);
+
+	constexpr bool selfTest = true;
+	bool initialized = imu.initialize(selfTest);
+	while (!initialized) {
+		MODM_LOG_ERROR << "Initialization failed, retrying ...\n";
+		initialized = imu.initialize(selfTest);
+		modm::delay(500ms);
+	}
```

Comment:
A do-while loop could make this more compact?

Reply:
I don't think that would be more compact. I neither want to do the wait nor print the error message if the first pass succeeds. The do-while loop would need another if on the inside to express the same logic.

Reply:
What about:
```suggestion
	while (not imu.initialize(selfTest)) 
	{
		MODM_LOG_ERROR << "Initialization failed, retrying ...\n";
		modm::delay(500ms);
	}
```